models/components.go

```diff
@@ -0,0 +1,539 @@
+		return nil
+	}
+	var ba []byte
+	switch v := val.(type) {
+	case []byte:
+		ba = v
+	default:
+		return errors.New(fmt.Sprint("Failed to unmarshal ResourceSelectors value:", val))
+	}
+	return json.Unmarshal(ba, rs)
+}
+
+func (rs ResourceSelectors) Value() (driver.Value, error) {
+	if len(rs) == 0 {
+		return []byte("[]"), nil
+	}
+	return json.Marshal(rs)
+}
+
+// GormDataType gorm common data type
+func (rs ResourceSelectors) GormDataType() string {
+	return "resourceSelectors"
+}
+
+// GormDBDataType gorm db data type
+func (ResourceSelectors) GormDBDataType(db *gorm.DB, field *schema.Field) string {
+	switch db.Dialector.Name() {
+	case SqliteType:
+		return jsonType
+	case PostgresType:
+		return jsonbType
+	case SQLServerType:
+		return nvarcharType
+	}
+	return ""
+}
+
+func (rs ResourceSelectors) GormValue(ctx context.Context, db *gorm.DB) clause.Expr {
+	data, _ := json.Marshal(rs)
+	return gorm.Expr("?", string(data))
+}
+
+type ComponentChecks []ComponentCheck
+
+type ComponentCheck struct {
+	Selector ResourceSelector `json:"selector,omitempty"`
+	Inline   *CanarySpec      `json:"inline,omitempty"`
+}
+
+func (cs ComponentChecks) Value() (driver.Value, error) {
+	if len(cs) == 0 {
+		return []byte("[]"), nil
+	}
+	return json.Marshal(cs)
+}
+
+func (cs *ComponentChecks) Scan(val interface{}) error {
+	if val == nil {
+		*cs = ComponentChecks{}
+		return nil
+	}
+	var ba []byte
+	switch v := val.(type) {
+	case []byte:
+		ba = v
+	default:
+		return errors.New(fmt.Sprint("Failed to unmarshal componentChecks value:", val))
+	}
+	return json.Unmarshal(ba, cs)
+}
+
+// GormDataType gorm common data type
+func (cs ComponentChecks) GormDataType() string {
+	return "componentChecks"
+}
+
+// GormDBDataType gorm db data type
+func (ComponentChecks) GormDBDataType(db *gorm.DB, field *schema.Field) string {
+	switch db.Dialector.Name() {
+	case SqliteType:
+		return jsonType
+	case PostgresType:
+		return jsonbType
+	case SQLServerType:
+		return nvarcharType
+	}
+	return ""
+}
+
+func (cs ComponentChecks) GormValue(ctx context.Context, db *gorm.DB) clause.Expr {
+	data, _ := json.Marshal(cs)
+	return gorm.Expr("?", string(data))
+}
+
+type ComponentRelationship struct {
+	ComponentID      uuid.UUID  `gorm:"column:component_id" json:"component_id,omitempty"`
+	RelationshipID   uuid.UUID  `gorm:"column:relationship_id" json:"relationship_id,omitempty"`
+	SelectorID       string     `gorm:"column:selector_id" json:"selector_id,omitempty"`
+	RelationshipPath string     `gorm:"column:relationship_path" json:"relationship_path,omitempty"`
+	CreatedAt        time.Time  `gorm:"column:created_at" json:"created_at,omitempty"`
+	UpdatedAt        time.Time  `gorm:"column:updated_at" json:"updated_at,omitempty"`
+	DeletedAt        *time.Time `gorm:"column:deleted_at" json:"deleted_at,omitempty"`
+}
+
+func (cr ComponentRelationship) TableName() string {
+	return "component_relationships"
+}
+
+type ConfigComponentRelationship struct {
```

Comment:
This came from `canary-checker/pkg/db`. Added the `gorm` tags & the last 3 fields.

Reply:
Gorm tags might not be required since the tags we added are the default anyway